tests/s25Main/campaign/testCampaignLuaFile.cpp

```diff
@@ -272,6 +290,93 @@ BOOST_AUTO_TEST_CASE(CampaignDescriptionLoadWithTranslation)
+                missionMapMask = {\"<RTTR_GAME>/GFX/PICS/WORLDMSK.LBM\", 0},\
+                marker         = {\"<RTTR_GAME>/DATA/IO/IO.DAT\", 231},\
+                conquered      = {\"<RTTR_GAME>/DATA/IO/IO.DAT\", 232},\
+                backgroundOffset = {64, 70},\
+                disabledColor = 0x70000000,\
+                missionSelectionInfos = {\
+                    {0xffffff00, 243, 97},\
+                    {0xffaf73cb, 55, 78},\
+                    {0xff008fc3, 122, 193}\
+                }\
+            }\
+        }";
+
+        file << "function getRequiredLuaVersion() return 2 end";
+    }
+
+    CampaignDescription desc;
+    CampaignDataLoader loader(desc, tmp.get());
+    BOOST_TEST_REQUIRE(loader.Load());
+
+    // campaign description
+    BOOST_TEST(desc.version == "1");
+    BOOST_TEST(desc.author == "Max Meier");
+    BOOST_TEST(desc.name == "Meine Kampagne");
+    BOOST_TEST(desc.shortDescription == "Sehr kurze Beschreibung");
+    BOOST_TEST(desc.longDescription == "Das ist die lange Beschreibung");
+    BOOST_TEST(desc.image == "<RTTR_GAME>/GFX/PICS/WORLD.LBM");
+    BOOST_TEST(desc.maxHumanPlayers == 1u);
+    BOOST_TEST(desc.difficulty == "easy");
+
+    // maps
+    BOOST_TEST(desc.getNumMaps() == static_cast<size_t>(3));
+    BOOST_TEST(desc.getMapFilePath(0) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/DATA/MAPS/dessert0.WLD"));
+    BOOST_TEST(desc.getMapFilePath(1) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/DATA/MAPS/dessert1.WLD"));
+    BOOST_TEST(desc.getMapFilePath(2) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/DATA/MAPS/dessert2.WLD"));
+    BOOST_TEST(desc.getLuaFilePath(0) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/CAMPAIGNS/ROMAN/dessert0.lua"));
+    BOOST_TEST(desc.getLuaFilePath(1) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/CAMPAIGNS/ROMAN/dessert1.lua"));
+    BOOST_TEST(desc.getLuaFilePath(2) == RTTRCONFIG.ExpandPath("<RTTR_GAME>/CAMPAIGNS/ROMAN/dessert2.lua"));
+
+    // selection map
+    auto selectionMap = desc.getSelectionMapData();
+    BOOST_TEST(selectionMap->background.filePath == "<RTTR_GAME>/GFX/PICS/SETUP990.LBM");
+    BOOST_TEST(selectionMap->background.index == 0u);
+    BOOST_TEST(selectionMap->map.filePath == "<RTTR_GAME>/GFX/PICS/WORLD.LBM");
+    BOOST_TEST(selectionMap->map.index == 0u);
+    BOOST_TEST(selectionMap->missionMapMask.filePath == "<RTTR_GAME>/GFX/PICS/WORLDMSK.LBM");
+    BOOST_TEST(selectionMap->missionMapMask.index == 0u);
+    BOOST_TEST(selectionMap->marker.filePath == "<RTTR_GAME>/DATA/IO/IO.DAT");
+    BOOST_TEST(selectionMap->marker.index == 231u);
+    BOOST_TEST(selectionMap->conquered.filePath == "<RTTR_GAME>/DATA/IO/IO.DAT");
+    BOOST_TEST(selectionMap->conquered.index == 232u);
+    BOOST_TEST(selectionMap->mapOffsetInBackground.x == 64);
+    BOOST_TEST(selectionMap->mapOffsetInBackground.y == 70);
+    BOOST_TEST(selectionMap->disabledColor == 0x70000000u);
+    BOOST_TEST(selectionMap->missionSelectionInfos.size() == static_cast<size_t>(3));
```

Comment:
I guess you were avoiding the signed-unsigned-mismatch warning? More readable:

```suggestion
    BOOST_TEST(selectionMap->missionSelectionInfos.size() == 3u);
```